### PR TITLE
Added information about Data Compaction

### PR DESCRIPTION
**Pull request by mikeburek**
Referenced the Microsoft blog post annoucing it: https://blog.fabric.microsoft.com/en-US/blog/announcing-automatic-data-compaction-for-fabric-warehouse/

I added the blog post to the reference section, but not sure if that is allowed in the documentation style.

# Thank you for contributing to Microsoft Fabric documentation

## Fill out these items before submitting your pull request:

If you are working internally at Microsoft:
**Provide a link to an Azure DevOps Boards work item that tracks this feature/update.**

>
>

**Who is your primary Skilling team contact?** \@mention them individually tag them and let them review the PR before signing off.

>
>

## For internal Microsoft contributors, check off these quality control items as you go

- [ ] 1. **Check the Acrolinx report:** Make sure your Acrolinx Total score is **above 80 minimum** (higher is better) and with no spelling issues. Acrolinx ensures we are providing consistent terminology and using an appropriate voice and tone, and helps with localization.

- [ ] 2. **Successful build with no warnings or suggestions**: Review the build status to make sure **all files are green** (Succeeded).

- [ ] 3. **Preview the pages:**: Click each **Preview URL** link to view the rendered HTML pages on the review.learn.microsoft.com site to check the formatting and alignment of the page. Scan the page for overall formatting, and look at the parts you edited in detail.

- [ ] 4. **Check the Table of Contents:** If you are adding a new markdown file, make sure it is linked from the table of contents.

- [ ] 5. **#sign-off to request PR review and merge**: Once the pull request is finalized and ready to be merged, indicate so by typing `#sign-off` in a new comment in the Pull Request. If you need to cancel that sign-off, type `#hold-off` instead. *Signing off means the document can be published at any time.* Note, this is a formatting and standards review, not a technical review.

## Merge and publish
- After you `#sign-off`, there is a separate PR Review team that will review the PR and describe any necessary feedback before merging. 
- The review team will use the comments section in the PR to provide feedback if changes are needed. Address any blocking issues and sign off again to request another review.
- Once all feedback is resolved, you can `#sign-off` again. The PR Review team reviews and merges the pull request into the specified branch (usually the main branch or a release- branch).
- From the main branch, the change is merged into the live branch several times a day to publish it to the public learn.microsoft.com site.

### CODE REVIEW

File: docs/data-warehouse/guidelines-warehouse-performance.md [hunk 1]
```diff
@@ -103,3 +109,4 @@ For information and recommendations on performance of the [!INCLUDE [fabric-se](
 - [T-SQL surface area](tsql-surface-area.md)
 - [Tables in data warehouse](tables.md)
 - [Caching in Fabric data warehousing](caching.md)
+- [Announcing: Automatic Data Compaction for Fabric Warehouse](https://blog.fabric.microsoft.com/en-US/blog/announcing-automatic-data-compaction-for-fabric-warehouse/)
```
Comment (WilliamDAssafMSFT) [inline docs/data-warehouse/guidelines-warehouse-performance.md:112]:
```suggestion
```

File: docs/data-warehouse/guidelines-warehouse-performance.md [hunk 2]
```diff
@@ -94,6 +94,12 @@ For supported data types and more information, see [data types](data-types.md#au
 
 For information and recommendations on performance of the [!INCLUDE [fabric-se](includes/fabric-se.md)], see [SQL analytics endpoint performance considerations](sql-analytics-endpoint-performance.md).
 
+## Data Compaction
+
+Data compaction consolidates smaller Parquet files into fewer, larger files, which optimizes read operations. This process also helps in efficiently managing deleted rows by eliminating them from immutable Parquet files. The data compaction process involves re-writing tables or segments of tables into new Parquet files that are optimized for performance.
+
+The data compaction process is seamlessly integrated into the Data Warehouse engine. As queries are executed, the system identifies tables that could benefit from compaction and performs necessary evaluations. There is not a manual way to trigger data compaction.
+
```
Comment (WilliamDAssafMSFT) [inline docs/data-warehouse/guidelines-warehouse-performance.md:102]:
```suggestion
## Data Compaction

Data compaction consolidates smaller Parquet files into fewer, larger files, which optimizes read operations. This process also helps in efficiently managing deleted rows by eliminating them from immutable Parquet files. The data compaction process involves re-writing tables or segments of tables into new Parquet files that are optimized for performance. For more information, see [Blog: Automatic Data Compaction for Fabric Warehouse](https://blog.fabric.microsoft.com/blog/announcing-automatic-data-compaction-for-fabric-warehouse/).

The data compaction process is seamlessly integrated into the warehouse. As queries are executed, the system identifies tables that could benefit from compaction and performs necessary evaluations. There is no manual way to trigger data compaction.

```

**Review by WilliamDAssafMSFT — CHANGES_REQUESTED** [review]
Thanks @Mikeburek for taking the time to submit this PR! Good catch. I made some edits, please accept the suggested commits. Then we can get this merged.

**Comment by mikeburek** [comment]
Thanks @WilliamDAssafMSFT for your expertise in the edits! I think I accepted all the changes. Let me know if I missed something.

**Comment by WilliamDAssafMSFT** [comment]
#sign-off

**Comment by WilliamDAssafMSFT** [comment]
#sign-off

**Pull request merged**